be/src/vec/exec/format/orc/vorc_reader.cpp

```diff
@@ -558,190 +562,316 @@ std::tuple<bool, orc::Literal> convert_to_orc_literal(const orc::Type* type, con
+        std::vector<orc::Literal> literals;
+        orc::PredicateDataType predicate_type;
+        for (size_t i = 1; i < expr->children().size(); ++i) {
+            DCHECK(expr->children()[i]->is_literal());
+            const auto* literal = static_cast<const VLiteral*>(expr->children()[i].get());
+            auto [valid, orc_literal, type] = _make_orc_leteral(slot_ref, literal);
+            if (!valid) {
+                return false;
+            }
+            literals.emplace_back(orc_literal);
+            predicate_type = type;
+        }
+        if (!literals.empty()) {
+            builder->startNot();
+            builder->in(slot_ref->expr_name(), predicate_type, literals);
+            builder->end();
+        }
+        break;
     }
-    if (index < predicates.size() - 1) {
-        bool can_build = build_search_argument(predicates, index + 1, builder);
-        if (!can_build) {
-            return false;
+    // is null and is not null is represented as function call
+    case TExprOpcode::INVALID_OPCODE: {
+        DCHECK(expr->node_type() == TExprNodeType::FUNCTION_CALL);
+        if (expr->fn().name.function_name == "is_null_pred") {
+            DCHECK(expr->children().size() == 1);
+            DCHECK(expr->children()[0]->is_slot_ref());
+            const auto* slot_ref = static_cast<const VSlotRef*>(expr->children()[0].get());
+            auto* slot = _tuple_descriptor->slots()[slot_ref->column_id()];
+            const auto* orc_type = _type_map[_col_name_to_file_col_name[slot->col_name()]];
+            const auto predicate_type = TYPEKIND_TO_PREDICATE_TYPE[orc_type->getKind()];
+            builder->isNull(slot_ref->expr_name(), predicate_type);
+        } else if (expr->fn().name.function_name == "is_not_null_pred") {
+            DCHECK(expr->children().size() == 1);
+            DCHECK(expr->children()[0]->is_slot_ref());
+            const auto* slot_ref = static_cast<const VSlotRef*>(expr->children()[0].get());
+            auto* slot = _tuple_descriptor->slots()[slot_ref->column_id()];
+            const auto* orc_type = _type_map[_col_name_to_file_col_name[slot->col_name()]];
+            const auto predicate_type = TYPEKIND_TO_PREDICATE_TYPE[orc_type->getKind()];
+            builder->startNot();
+            builder->isNull(slot_ref->expr_name(), predicate_type);
+            builder->end();
+        } else {
+            __builtin_unreachable();
         }
-        builder->end();
+        break;
+    }
+    default: {
+        // should not reach here, because _check_expr_can_push_down has already checked
+        __builtin_unreachable();
+    }
     }
     return true;
 }
 
-bool OrcReader::_init_search_argument(
-        std::unordered_map<std::string, ColumnValueRangeType>* colname_to_value_range) {
-    if ((!_enable_filter_by_min_max) || colname_to_value_range->empty()) {
+bool OrcReader::_init_search_argument(const VExprContextSPtrs& conjuncts) {
```

Comment:
For a certain query, the conjuncts is identical in all orc readers.
So I think we can just init search argument once and use the result for every orc reader?

Reply:
it is possible to cache searchArgument into member variables of OrcReader